resources/aws_ec2.go

```diff
@@ -858,16 +990,135 @@ func (obj *AwsEc2Res) snsDeleteTopic(topicArn string) error {
 	return nil
 }
 
+// snsSubscribe subscribes the endpoint to the sns topic.
+// Returning SubscriptionArn here is useless as it is still pending confirmation.
+func (obj *AwsEc2Res) snsSubscribe(endpoint string, topicArn string) error {
+	// subscribe to the topic
+	subInput := &sns.SubscribeInput{
+		Endpoint: aws.String(endpoint),
+		Protocol: aws.String(SnsSubscriptionProto),
+		TopicArn: aws.String(topicArn),
+	}
+	_, err := obj.snsClient.Subscribe(subInput)
+	if err != nil {
+		return err
+	}
+	log.Printf("%s: Created Subscription", obj)
+	return nil
+}
+
+// snsConfirmSubscription confirms the sns subscription.
+// Returning SubscriptionArn here is useless as it is still pending confirmation.
+func (obj *AwsEc2Res) snsConfirmSubscription(topicArn string, token string) error {
+	// confirm the subscription
+	csInput := &sns.ConfirmSubscriptionInput{
+		Token:    aws.String(token),
+		TopicArn: aws.String(topicArn),
+	}
+	_, err := obj.snsClient.ConfirmSubscription(csInput)
+	if err != nil {
+		return err
+	}
+	log.Printf("%s: Subscription Confirmed", obj)
+	return nil
+}
+
+// cweMakeEventPattern makes and encodes event patterns for cloudwatch rules.
+func (obj *AwsEc2Res) cweMakeEventPattern(source string, detailType string, detail []string) (string, error) {
+	pattern := cloudWatchRule{
+		Source:     []string{source},
+		DetailType: []string{detailType},
+		Detail: ruleDetail{
+			State: detail,
+		},
+	}
+	eventPattern, err := json.Marshal(pattern)
+	if err != nil {
+		return "", err
+	}
+	return string(eventPattern), nil
+}
+
+// cweMakeRule makes a cloud watch rule.
+func (obj *AwsEc2Res) cweMakeRule(name string, eventPattern string) error {
+	// make cloudwatch rule
+	putRuleInput := &cwe.PutRuleInput{
+		Name:         aws.String(name),
+		EventPattern: aws.String(eventPattern),
+	}
+	_, err := obj.cweClient.PutRule(putRuleInput)
+	if err != nil {
+		return err
+	}
+	log.Printf("%s: Created CloudWatch Rule", obj)
+	return nil
+}
+
+// cweDeleteRule removes the target and deletes the cloudwatch rule.
+func (obj *AwsEc2Res) cweDeleteRule(name string) error {
+	// delete the rule
+	drInput := &cwe.DeleteRuleInput{
+		Name: aws.String(name),
+	}
+	log.Printf("%s: Deleting CloudWatch Rule", obj)
```

Comment:
I need to change this, and the one in cweRemoveTarget to past tense like the other messages. Will leave it as is until you review.